compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/statements/BLangBlock.java

```diff
@@ -0,0 +1,66 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.ballerinalang.compiler.tree.statements;
+
+import org.ballerinalang.model.tree.NodeKind;
+import org.ballerinalang.model.tree.statements.BlockNode;
+import org.ballerinalang.model.tree.statements.BlockStatementNode;
+import org.wso2.ballerinalang.compiler.tree.BLangNodeVisitor;
+
+/**
+ * @since 0.94
+ */
+public class BLangBlock extends BLangStatement implements BlockNode {
```

Comment:
Can we not use `BLangBlockStmt`, `BlockStatementNode` with some modifications instead of introducing new nodes? They are the same right? IMO, with this change we're introducing two ways to do the same thing.

@pubudu91 thoughts?

Reply:
The reason to add new node is to specialize block statement as one type of statement. I think when considering BLangBlockStmt it keeps all types of statements.

Reply:
Logic was changed according to available nodes.(BLangBlockStmt, BlockStatementNode)